src/index.js

```diff
@@ -235,10 +239,373 @@ const maliciousSetApprovalForAll = document.getElementById(
+  sendButton,
+  createToken,
+  decimalUnitsInput,
+  watchAssets,
+  transferTokens,
+  approveTokens,
+  transferTokensWithoutGas,
+  approveTokensWithoutGas,
+  getEncryptionKeyButton,
+  encryptMessageInput,
+  encryptButton,
+  decryptButton,
+  ethSign,
+  personalSign,
+  personalSignVerify,
+  signTypedData,
+  signTypedDataVerify,
+  signTypedDataV3,
+  signTypedDataV3Verify,
+  signTypedDataV4,
+  signTypedDataV4Verify,
+  signPermit,
+  signPermitVerify,
+  siwe,
+  siweResources,
+  siweBadDomain,
+  siweBadAccount,
+  siweMalformed,
+  eip747WatchButton,
+  maliciousApprovalButton,
+  maliciousSetApprovalForAll,
+  maliciousERC20TransferButton,
+  maliciousRawEthButton,
+  maliciousPermit,
+  maliciousTradeOrder,
+  maliciousSeaport,
+];
+
+// Buttons that are available after initially connecting an account
+const initialConnectedButtons = [
+  deployButton,
+  deployNFTsButton,
+  deployERC1155Button,
+  sendButton,
+  deployFailingButton,
+  deployMultisigButton,
+  createToken,
+  decimalUnitsInput,
+  personalSign,
+  signTypedData,
+  getEncryptionKeyButton,
+  ethSign,
+  personalSign,
+  signTypedData,
+  signTypedDataV3,
+  signTypedDataV4,
+  signPermit,
+  siwe,
+  siweResources,
+  siweBadDomain,
+  siweBadAccount,
+  siweMalformed,
+  eip747WatchButton,
+];
+
+/**
+ * Provider
+ */
+
+const providerDetails = [];
+let provider;
+let accounts = [];
+let scrollToHandled = false;
+
+const isMetaMaskConnected = () => accounts && accounts.length > 0;
+
+// TODO: Need to align with @metamask/onboarding
+const isMetaMaskInstalled = () => provider && provider.isMetaMask;
+
+const detecEip6963 = () => {
+  window.addEventListener('eip6963:announceProvider', (event) => {
+    if (event.detail.info.uuid) {
+      eip6963Warning.hidden = true;
+      eip6963Section.hidden = false;
+
+      handleNewProviderDetail(event.detail);
+    }
+  });
+
+  window.dispatchEvent(new Event('eip6963:requestProvider'));
+};
+
+const setActiveProviderDetail = (providerDetail) => {
+  closeProvider();
+  provider = providerDetail.provider;
+  initializeProvider();
+
+  const { uuid, name, icon } = providerDetail.info;
+  activeProviderUUIDResult.innerText = uuid;
+  activeProviderNameResult.innerText = name;
+  activeProviderIconResult.innerHTML = icon
+    ? `<img src="${icon}" height="90" width="90" />`
+    : '';
+};
+
+const setActiveProviderDetailWindowEthereum = () => {
+  const providerDetail = {
+    info: {
+      uuid: '',
+      name: 'window.ethereum',
+      icon: '',
+    },
+    provider: window.ethereum,
+  };
+  setActiveProviderDetail(providerDetail);
+};
+
+const existsProviderDetail = (newProviderDetail) => {
+  const existingProvider = providerDetails.find(
+    (providerDetail) =>
+      providerDetail.info &&
+      newProviderDetail.info &&
+      providerDetail.info.uuid === newProviderDetail.info.uuid,
+  );
+
+  if (existingProvider) {
+    if (existingProvider.info.name !== newProviderDetail.info.name) {
```

Comment:
sorry, this is something i should have done. Can we check `rdns` and `icon` values too? the point of this was to make sure that when we received the a `uuid` in ProviderInfo that we have already seen before that it is exactly the same ProviderInfo, otherwise something funky might be going on

Reply:
Is [701ff7e](https://github.com/MetaMask/test-dapp/pull/263/commits/701ff7e8f9ea50bce1ee651934406a75878c5813) what you had it mind?

Reply:
works for me. thank you